### PR TITLE
feature: Global State with Context API for data Timeframes

### PR DESCRIPTION
**Pull request by peeyalk**
## Evidence

For mobile screens;

|Before|After|
|---|---|
|---|---|

For desktop screens;

|Before|
|---|
|---|

|After|
|---|
|---|

### Environment Variables Setting

Do you require registration for the following console screens beside the source code? If so, please attach evidence of registration to each of the console screens below.

- GitHub Actions: 
- Vercel Hosting: 
- Firebase Console: 
- Google Cloud Platform: 
- Asana Developer Console: 
- Slack Developer Console: 

## Performance

For desktop screen,

- DOM Content Loaded Speed: x.xx (s) --> x.xx (s), -xxx (ms)
- Finish: x.xx (s) --> x.xx (s), -xxx (ms)

|Chrome Developer Console Network Tab (Before)|
|---|
|---|

|Chrome Developer Console Network Tab (After)|
|---|
|---|

### CODE REVIEW

File: components/buttons/SpecifyPeriodFixedTerm.tsx [hunk 1]
```diff
@@ -1,18 +1,39 @@
+import { useEffect, useState } from "react";
+import { handleSignoutClick } from "../../services/googleCalendar.client";
+
 interface SpecifyPeriodFixedTermTypes {
-  bgColor?: string;
+  currentTimeframe?: {
+    timeframe: {
+      label: string
+    },
+    setTimeFrame: any // func dispatchSetState
```
[the remark this answers is not in the document]
Comment (peeyalk) [reply components/buttons/SpecifyPeriodFixedTerm.tsx]: Also this

File: constants/timestampButtonList.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+[
+    {
+        "label": "Full",
+        "disabled": false,
```
[the remark this answers is not in the document]
Comment (peeyalk) [reply constants/timestampButtonList.json:4]: No it's is set from the global state

File: components/buttons/SpecifyPeriodFixedTerm.tsx [hunk 3]
```diff
@@ -1,18 +1,39 @@
...
+  };
   disabled?: boolean;
```
[the remark this answers is not in the document]
Comment (hiroshinishio) [reply components/buttons/SpecifyPeriodFixedTerm.tsx:10]: What do you think about this?

[the remark this answers is not in the document]
Comment (peeyalk) [reply components/buttons/SpecifyPeriodFixedTerm.tsx:10]: @hiroshinishio 
Yes it is managed in the component life cycle

[the remark this answers is not in the document]
Comment (hiroshinishio) [reply components/buttons/SpecifyPeriodFixedTerm.tsx:10]: I mean this 'disabled' type is unnecessary I think so you better remove it??

**Review by hiroshinishio — APPROVED** [review]

**Pull request merged**